persist/sqlite/blocks.go

```diff
@@ -15,7 +15,9 @@ func (s *Store) Block(id types.BlockID) (result explorer.Block, err error) {
 		var v2Height uint64
 		var v2Commitment types.Hash256
 		err := tx.QueryRow(`SELECT parent_id, nonce, timestamp, height, leaf_index, v2_height, v2_commitment FROM blocks WHERE id = ?`, encode(id)).Scan(decode(&result.ParentID), decode(&result.Nonce), decode(&result.Timestamp), &result.Height, decode(&result.LeafIndex), decodeNull(&v2Height), decodeNull(&v2Commitment))
-		if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return explorer.ErrNoBlock
```

Comment:
There should be a unit test for `Block` that you can extend to check if this particular error is returned correctly